### PR TITLE
Check if the amount of predicted checks match the amount of checks done, print warning if not.

### PR DESCRIPTION
**Pull request by marcel-baur**
Regards (#409). 
Are there any cases where this issue occurs?  If so, I could add a test case. 
#### ✔ Checklist:

- [x] Commit messages have been written in [Conventional Commits](https://www.conventionalcommits.org) format;
- [x] This PR has been added to [CHANGELOG.md](https://github.com/dotenv-linter/dotenv-linter/blob/master/CHANGELOG.md) (at the top of the list);
- [ ] Tests for the changes have been added (for bug fixes / features);

### CODE REVIEW

File: src/common/output/fix.rs [hunk 1]
```diff
@@ -63,4 +63,13 @@ impl FixOutput {
...
         println!("Nothing to fix");
     }
+
+    /// Prints not all warnings fixed message
+    pub fn print_not_all_warnings_fixes(&self) {
+        if self.is_quiet_mode {
+            return;
+        }
+
+        println!("Could not fix all warnings!");
```
Comment (mgrachev) [inline src/common/output/fix.rs:73]:
```suggestion
        println!("Could not fix all warnings".red().bold());
```

File: src/lib.rs [hunk 2]
```diff
@@ -68,9 +68,17 @@ pub fn fix(args: &clap::ArgMatches, current_dir: &Path) -> Result<(), Box<dyn Er
...
+        let mut fixes_done = 0;
+        if !result.is_empty() {
+            fixes_done = fixes::run(&mut result, &mut lines, &skip_checks);
+        }
+        // Compare amount of fixes done to amount of expected fixes. Print warning if they differ.
+        if fixes_done < expected_fixes {
+            output.print_not_all_warnings_fixes();
+        }
+        // write results to file
+        if !result.is_empty() && fixes_done > 0 {
```
Comment (mgrachev) [inline src/lib.rs:81]: I think we can simplify this code 🤔
```suggestion
        if result.is_empty() {
            continue;
        }

        let fixes_done = fixes::run(&mut result, &mut lines, &skip_checks);
        if fixes_done != result.len() {
            output.print_not_all_warnings_fixes();
        }
        if fixes_done > 0 {
```

**Review by mgrachev — COMMENTED** [review]
@marcel-baur 👋 Thank you for your contribution 🦀

I have left some comments. Please take a look at them 👀

**Comment by mgrachev** [comment]
Also, please do this one ⬇️
> - [ ] This PR has been added to [CHANGELOG.md](https://github.com/dotenv-linter/dotenv-linter/blob/master/CHANGELOG.md) (at the top of the list);

And please fix CI - all checks have to be green ✅
> Linters / linters / commitlint (pull_request) Failing after 13s — linters / commitlint

**Comment by mgrachev** [comment]
Please do this one ⬇️
> - [ ] This PR has been added to [CHANGELOG.md](https://github.com/dotenv-linter/dotenv-linter/blob/master/CHANGELOG.md) (at the top of the list);

**Comment by marcel-baur** [comment]
> Please do this one ⬇️
> 
> > * [ ]  This PR has been added to [CHANGELOG.md](https://github.com/dotenv-linter/dotenv-linter/blob/master/CHANGELOG.md) (at the top of the list);

I think I have already done this with a change request, should I do that within this pull request as well?

**Comment by mgrachev** [comment]
> > Please do this one ⬇️
> > > * [ ]   This PR has been added to [CHANGELOG.md](https://github.com/dotenv-linter/dotenv-linter/blob/master/CHANGELOG.md) (at the top of the list);
> 
> I think I have already done this with a change request, should I do that within this pull request as well?

Yes, please.

File: src/lib.rs [hunk 3]
```diff
@@ -68,9 +68,14 @@ pub fn fix(args: &clap::ArgMatches, current_dir: &Path) -> Result<(), Box<dyn Er
...
         let mut result = checks::run(&lines, &skip_checks);
-
-        // run fixers & write results to file
-        if !result.is_empty() && fixes::run(&mut result, &mut lines, &skip_checks) > 0 {
+        if result.is_empty() {
+            continue;
+        }
+        let fixes_done = fixes::run(&mut result, &mut lines, &skip_checks);
+        if fixes_done != result.len() {
+            output.print_not_all_warnings_fixes();
```
Comment (DDtKey) [inline src/lib.rs:76]:
```suggestion
            output.print_not_all_warnings_fixed();
```

**Review by DDtKey — DISMISSED** [review]
Thank you for your contribution! 👍 

As far as I can see, only my comment on the changelog remains. But it's okay I think

**Review by DDtKey — APPROVED** [review]

**Review by mgrachev — APPROVED** [review]

**Comment by mgrachev** [comment]
@marcel-baur Thank you for your help 👍

🙏 If it’s not difficult for you, please support the project - star on GitHub ⭐️

**Pull request merged**